fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java

```diff
@@ -1214,15 +1231,65 @@ private Money processAllocationsHorizontally(LoanTransaction loanTransaction, Mo
                             for (LoanRepaymentScheduleInstallment inAdvanceInstallment : inAdvanceInstallments) {
                                 Set<LoanCharge> inAdvanceInstallmentCharges = getLoanChargesOfInstallment(charges, inAdvanceInstallment,
                                         firstNormalInstallmentNumber);
+
                                 // Adjust the portion for the last installment
                                 if (inAdvanceInstallment.equals(inAdvanceInstallments.get(numberOfInstallments - 1))) {
                                     evenPortion = evenPortion.add(balanceAdjustment);
                                 }
                                 LoanTransactionToRepaymentScheduleMapping loanTransactionToRepaymentScheduleMapping = getTransactionMapping(
                                         transactionMappings, loanTransaction, inAdvanceInstallment, currency);
-                                paidPortion = processPaymentAllocation(paymentAllocationType, inAdvanceInstallment, loanTransaction,
-                                        evenPortion, loanTransactionToRepaymentScheduleMapping, inAdvanceInstallmentCharges, balances,
-                                        LoanRepaymentScheduleInstallment.PaymentAction.PAY);
+
+                                Loan loan = loanTransaction.getLoan();
+                                if (transactionCtx instanceof ProgressiveTransactionCtx ctx && loan.isInterestBearing()
+                                        && loan.getLoanProductRelatedDetail().isInterestRecalculationEnabled()) {
+                                    ProgressiveLoanInterestScheduleModel model = ctx.getModel();
+                                    LocalDate transactionDate = loanTransaction.getTransactionDate();
+                                    LocalDate payDate = inAdvanceInstallment.getFromDate().isAfter(transactionDate)
+                                            ? inAdvanceInstallment.getFromDate()
+                                            : transactionDate;
+                                    ProgressiveLoanInterestRepaymentModel payableDetails = emiCalculator
+                                            .getPayableDetails(model, inAdvanceInstallment.getDueDate(), payDate).orElseThrow();
+
+                                    switch (paymentAllocationType) {
+                                        case IN_ADVANCE_INTEREST -> {
+                                            log.info("{} - asking calculator for paydate {}, interest: {}",
```

Comment:
Probably this is a bit verbose on info, do you think it would make sense to move it to debug level?